couchdb/templates/clustersetup.yaml

```diff
@@ -0,0 +1,76 @@
+{{ if .Values.clusterSetup }}
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-setup-script
+  labels:
+    app: {{ template "couchdb.name" . }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+data:
+  setup.sh: |
+    set -e
+    BASE_URL=http://{{ template "couchdb.fullname" . }}:5984
+    set -x
+
+    echo "Waiting for CouchDB service to start..."
+    until curl --silent -max-time 5 --head --fail "${BASE_URL}/_up"; do
```

Comment:
it's possible `_up` requires authentication. You likely need to copy the logic from https://github.com/apache/couchdb-helm/blob/master/couchdb/templates/statefulset.yaml#L87 to deal with this

Reply:
ah sorry - I think your approach is neater.

Reply:
I've made the update using the curl config file option, but if you tell me that's not useful in a K8s context, I'll update to use CLI flags.